pkg/appmgmt/sparkoperator/spark.go

```diff
@@ -124,3 +128,21 @@ func (os *Manager) deleteApplication(obj interface{}) {
 	log.Logger().Info("spark app deleted", zap.Any("SparkApplication", app))
 	os.amProtocol.NotifyApplicationComplete(app.Status.SparkApplicationID)
 }
+
```

Comment:
We need to ensure that the YuniKorn label takes precedence if specified. Otherwise we create a security hole if we have an environment where the username label is forced by policy (such as an admission controller that checks security credentials). Allowing Spark to override this is wrong.